torchdynamo/variables/functions.py

```diff
@@ -78,8 +79,17 @@ def closure_vars(self, tx):
 class UserFunctionVariable(BaseUserFunctionVariable):
```

Comment:
You're making some significant changes to this object, could we make a separate `ConstantFunctionVariable` type that we create in the builder if the right attributes are present?

Reply:
Well, `UserFunctionVariable` can be const, but so can `UserMethodVariable` and it felt a little gross to fork it into ConstantUserFunctionVariable and ConstantUserMethodVariable, but I am not strongly opposed.